pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -503,13 +503,22 @@ public CompletableFuture<Optional<String>> getOwnerAsync(String serviceUnit) {
             case Splitting -> {
                 return CompletableFuture.completedFuture(Optional.of(data.sourceBroker()));
             }
-            case Assigning, Releasing -> {
+            case Assigning -> {
                 return deferGetOwnerRequest(serviceUnit).whenComplete((__, e) -> {
                     if (e != null) {
                         ownerLookUpCounters.get(state).getFailure().incrementAndGet();
                     }
-                }).thenApply(
-                        broker -> broker == null ? Optional.empty() : Optional.of(broker));
+                }).thenApply(broker -> broker == null ? Optional.empty() : Optional.of(broker));
+            }
+            case Releasing -> {
+                if (isTransferCommand(data)) {
+                    return deferGetOwnerRequest(serviceUnit).whenComplete((__, e) -> {
+                        if (e != null) {
+                            ownerLookUpCounters.get(state).getFailure().incrementAndGet();
+                        }
+                    }).thenApply(broker -> broker == null ? Optional.empty() : Optional.of(broker));
+                }
+                return CompletableFuture.completedFuture(Optional.empty());
```

Comment:
What happens if there are many lookups at this Releasing state(non-transfer)?

Reply:
Also, we have the leader broker who should watch any stuck states and fix the stuck state. Do we know why the monitor didn't kick in this case and didin't complete the future(maybe because of the delay)?

Reply:
The monitor will clean up this state, but we should unblock it for admin API since we don't want the admin API timeout.

Reply:
Then, isn't your change in UnloadManager.handleEvent sufficient to return the admin api?

In the test env, I feel like we can configure the monitor frequency small(like 1 sec) to fix the stuck state quicker.

Reply:
> Then, isn't your change in UnloadManager.handleEvent sufficient to return the admin api?
No, the `unloadServiceUnit` will timeout since the current is `Releasing` state.

You can see we are calling `closeServiceUnit` in `handleReleaseEvent` method, but in the `closeServiceUnit`, it will check the ownership.

So when it gets ownership, this operation will be blocked since the current state is `Releasing`.(The `Free` state message has not been sent to the channel yet)

Reply:
But still, doesn't the monitor fix the state and complete the getOwner future soon if we make the monitor delay 1sec in this test env? 
I am worried that returning the ownership early in this in-flight state could cause racing conditions.

Reply:
Or, can we skip the ownership check when the ML update fails(if extensible lb)? I think it can just get the ledger info again no matter what. When the monitor fixes the stuck state, it should clean the ledger again.

Or, could we expose a new internal getOwner api without deferring future(this should not be called by lookups)?

Reply:
> But still, doesn't the monitor fix the state and complete the getOwner future soon if we make the monitor delay 1sec in this test env?

We need to change the `inFlightStateWaitingTimeInMillis` and `ownershipMonitorDelayTimeInSecs` to 1sec, I wonder if they will cause the same normal in-flight state to get overridden. Maybe we can delay 5sec in the test. 

Another issue is it will override the `Releasing` -> `Own`, the unload result is not expected.

> I am worried that returning the ownership early in this in-flight state could cause racing conditions.

I see the ownership cache(old logic) will mark the current state as inactive, does it have the same issue?

> Also, it appears that the same issue could happen from transfers too.

Yes, the transfer has the same issue.

> Or, can we skip the ownership check when the ML update fails(if extensible lb)? I think it can just get the ledger info again no matter what. When the monitor fixes the stuck state, it should clean the ledger again.

Yes, it can work for tests, but still, I'm not sure if that is this unload result is correct(`Releasing` -> `Own`).

> Or, could we expose a new internal getOwner api without deferring future(this should not be called by lookups)?

The `getOwner` will use the table view state directly?

Reply:
> Or, can we skip the ownership check when the ML update fails(if extensible lb)? I think it can just get the ledger info again no matter what. When the monitor fixes the stuck state, it should clean the ledger again.

>> Yes, it can work for tests, but still, I'm not sure if that is this unload result is correct(Releasing -> Own).

As long as the source unblocks and returns error, I think we are good here.

I believe we need a separate PR to fix the case when the ledger info update fails with BadVersion. This shouldn't happen as the only source should own the topic.

Reply:
> I believe we need a separate PR to fix the case when the ledger info update fails with BadVersion. This shouldn't happen as the only source should own the topic.

Okay, updated.

Reply:
thanks!